Domain suggestions page layout redesign

hambai opened the pull request:
Implement the new domain suggestions layout screen design

#### Changes proposed in this Pull Request

* Update teh domain suggestions page layout

The updated screens looks like this (add a domain, NUX and domain-only flows):

Before:
<img width="1008" alt="Screenshot 2021-08-12 at 8 19 01" src="https://user-images.githubusercontent.com/1355045/129142264-f5c136d0-ad09-47ae-8c96-db0d9ed68d11.png">
<img width="962" alt="Screenshot 2021-08-12 at 8 19 20" src="https://user-images.githubusercontent.com/1355045/129142268-4136a88b-fc03-492c-b012-b0369e454576.png">
<img width="1074" alt="Screenshot 2021-08-12 at 8 19 37" src="https://user-images.githubusercontent.com/1355045/129142270-b6d8732f-3131-4b26-a904-368adc4bad37.png">

After:
<img width="1010" alt="Screenshot 2021-08-12 at 8 14 27" src="https://user-images.githubusercontent.com/1355045/129142284-ecde3719-ba68-4c36-af37-6032ec3a5422.png">
<img width="975" alt="Screenshot 2021-08-12 at 8 15 14" src="https://user-images.githubusercontent.com/1355045/129142293-099ec413-5509-45a2-b049-9a4061517995.png">
<img width="1084" alt="Screenshot 2021-08-16 at 22 07 42" src="https://user-images.githubusercontent.com/1355045/129616739-f821d60e-aa56-4fab-bf19-320771aeb128.png">

#### Testing instructions

* Open this branch and verify that the domain suggestions screen layout is the new layout (search field and the filter buttons are separated and basically everything is inside one big Card)

hambai committed:
Initial domain suggestions page layout change

We'll need to rebase this once the other things are merged

hambai committed:
Fix the styles for the domain-only flow

hambai committed:
Fix the filter button to be rendered outside the input

hambai committed:
* remove the active filter button box-shadow

poligilad-auto reviewed (changes requested):
1. Domain-only: there is a layout issue.

![image](https://user-images.githubusercontent.com/83961704/129347483-84bb3b5d-3b8f-42f5-9f76-7a16c2d00234.png)

2. There is some extra spacing here in mobile view:

![image](https://user-images.githubusercontent.com/83961704/129354404-b1e1d579-0f44-47ae-8abf-104804fe5f3a.png)

3. Show more results: remove the last line and fix the spacing to be identical to the spacing above "Show more results". Like the pic below.

![image](https://user-images.githubusercontent.com/83961704/129354670-ce438379-ba13-4e36-92da-e38e9cd556a5.png)

4. The "sticky version" of the search bar. It currently looks a bit strange. Could we add some styling so it looks like the version below, white background, extra padding to the top, add a gradient div below to separate it from the content.

![image](https://user-images.githubusercontent.com/83961704/129355449-0ec23396-cad7-4f90-b888-744f9675dc9d.png)

5. Mobile: let's make the padding for the whole area 16px instead of 24px, that gives more space for the content.

6. Very small thing: there is this 1 px difference between the search bar and the "Filter" button. You can notice it when you see the sticky version.

hambai committed:
* fix the mobile padding
* fix the pagination margin

hambai committed:
* fix 1px height diff

hambai committed:
* fix the select button width

hambai commented:
> 1. Domain-only: there is a layout issue.
We'll fix this in a separate PR - I've added a card to our board to handle it

> 1. There is some extra spacing here in mobile view:

Hmm... I couldn't reproduce this. We'll hopefully notice if it happens again

> 1. Show more results: remove the last line and fix the spacing to be identical to the spacing above "Show more results". Like the pic below.

Fixed in the current PR.

> 1. The "sticky version" of the search bar. It currently looks a bit strange. Could we add some styling so it looks like the version below, white background, extra padding to the top, add a gradient div below to separate it from the content.

More complicated than I thought - will fix it in a separate PR

> 1. Mobile: let's make the padding for the whole area 16px instead of 24px, that gives more space for the content.

Fixed in the current PR.

> 2. Very small thing: there is this 1 px difference between the search bar and the "Filter" button. You can notice it when you see the sticky version.

Fixed in the current PR.

hambai committed:
* fix pagination button padding
